File: Programmers_python/올바른_괄호.py
def solution(s):
    stack = []

    for c in s:
        if c == ')':
            if len(stack) == 0:
                return False
            top = stack.pop()
            if top != '(':
                return False

        elif c == '(':
            stack.append(c)

    if len(stack) > 0:
        return False
    return True

File: Programmers_js/올바른 괄호.js
function solution(s){
    let answer = true;
    const stack = [];
    
    for(let i=0, len=s.length; i<len; i++) {
        if(s[i] === "(") {
            stack.push("(");
        } else if(s[i] === ")") {
            if(stack.length == 0) {
                answer = false;
                break;
            }
            
            if(stack[stack.length-1] === "(") {
                stack.pop();
                continue;
            } else if(stack[stack.length-1] === ")") {
                answer = false;
                break;
            }
        }
    }

    if(stack.length > 0) {
        answer = false;
    }
    
    return answer;
}
